### PR TITLE
feat(v2): allow creation of date fields in admin form builder

### LINKED ISSUE #3399: [React/Builder] Implement Date field drawer and preview

### PR DESCRIPTION
**Pull request by karrui**
## Problem

This PR adds ability to create attachment fields in admin form builder screen. 

Closes #3399  

## Solution

**Features**:

- allow creation of date fields in admin form builder screen

## Before & After Screenshots
New stories added for EditDate component.

### CODE REVIEW

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditDate/EditDate.tsx [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+}
+
+const transformDateFieldToEditForm = (field: DateFieldBase): EditDateInputs => {
+  const nextValidationOptions = {
+    selectedDateValidation:
+      field.dateValidation.selectedDateValidation ?? ('' as const),
+    customMaxDate:
+      (!!field.dateValidation.selectedDateValidation &&
+        transformDateToShortIsoString(field.dateValidation.customMaxDate)) ||
+      ('' as const),
```
Comment (timotheeg) [inline frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditDate/EditDate.tsx:53]: Considering what this does, a ternary might be clearer, and same for `customMinDate` below
```suggestion
    customMaxDate: (field.dateValidation.selectedDateValidation
        ? transformDateToShortIsoString(field.dateValidation.customMaxDate)
        : ('' as const)),
```

**Review by timotheeg — APPROVED** [review]
lgtm

File: frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditDate/EditDate.tsx (hunk 1 above)
Comment (karrui) [reply frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditDate/EditDate.tsx:53]: transformDateToShortIsoString can return `null`. added an extra nullish check, think your suggestion with a nullish check is still clearer

**Review by timotheeg — APPROVED** [review]
+1 again

**Pull request merged**